framework/include/utils/MathFVUtils.h

```diff
@@ -241,6 +243,90 @@ linearInterpolation(const T & value1,
   return coeffs.first * value1 + coeffs.second * value2;
 }
 
+/**
+ * Computes the harmonic mean (1/(gc/value1+(1-gc)/value2)) of Reals, RealVectorValues and
+ * RealTensorValues while accounting for the possibility that one or both of them are AD.
+ * For tensors, we use a component-wise mean instead of the matrix-inverse based option.
+ * @param value1 Reference to value1 in the (1/(gc/value1+(1-gc)/value2)) expression
+ * @param value2 Reference to value2 in the (1/(gc/value1+(1-gc)/value2)) expression
+ * @param fi Reference to the FaceInfo of the face on which the interpolation is requested
+ * @param one_is_elem Boolean indicating if the interpolation weight on FaceInfo belongs to the
+ * elementcorresponding to value1
+ * @return The interpolated value
+ */
+template <typename T1, typename T2>
+typename libMesh::CompareTypes<T1, T2>::supertype
+harmonicInterpolation(const T1 & value1,
+                      const T2 & value2,
+                      const FaceInfo & fi,
+                      const bool one_is_elem)
+{
+  // We check if the base values of the given template types match, if not we throw a compile-time
+  // error
+  static_assert(std::is_same<typename MetaPhysicL::RawType<T1>::value_type,
+                             typename MetaPhysicL::RawType<T2>::value_type>::value,
+                "The input values for harmonic interpolation need to have the same raw-value!");
+
+  // Fetch the interpolation coefficients, we use the exact same coefficients as for a simple
+  // weighted average
+  const auto coeffs = interpCoeffs(InterpMethod::Average, fi, one_is_elem);
+
+  // We check if the types are fit to compute the harmonic mean of. This is done compile-time
+  // using constexpr. We start with Real/ADReal which is straightforward if the input values are
+  // positive.
+  if constexpr (std::is_same<typename MetaPhysicL::RawType<T1>::value_type, Real>::value)
+  {
+    // The harmonic mean of mixed positive and negative numbers (and 0 as well) is not well-defined
+    // so we assert that the input values shall be positive.
+    mooseAssert(value1 > 0, "Input value 1 needs to be positive for harmonic interpolation!");
+    mooseAssert(value2 > 0, "Input value 2 needs to be positive for harmonic interpolation!");
+    return 1.0 / (coeffs.first / value1 + coeffs.second / value2);
+  }
+  // For vectors (ADRealVectorValue, VectorValue), we take the component-wise harmonic mean
+  else if constexpr (std::is_same<typename MetaPhysicL::RawType<T1>::value_type,
+                                  RealVectorValue>::value)
+  {
+    typename libMesh::CompareTypes<T1, T2>::supertype result;
+    for (const auto i : make_range(LIBMESH_DIM))
```

Comment:
```suggestion
    for (const auto i : make_range(Moose::dim))
```

Reply:
Is there a difference between moose::dim and libmesh_dim?

Reply:
`Moose::dim` is actually typed whereas `LIBMESH_DIM` is a macro